pkg/cmd/pipeline/list.go

```diff
@@ -16,51 +16,172 @@ package pipeline
-			ps, err := list(p)
-
+			output, err := cmd.LocalFlags().GetString("output")
 			if err != nil {
-				fmt.Fprintf(os.Stderr, "Failed to list pipelines from %s namespace\n", p.Namespace())
+				fmt.Fprint(os.Stderr, "Error: output option not set properly \n")
 				return err
 			}
 
-			printer, err := f.ToPrinter()
-			if err != nil {
-				return err
+			if output != "" {
+				return printPipelineListObj(cmd.OutOrStdout(), p, f)
 			}
-			return printer.PrintObj(ps, cmd.OutOrStdout())
+
+			return printPipelineDetails(cmd.OutOrStdout(), p)
 		},
 	}
 	f.AddFlags(c)
 
 	return c
 }
 
-func list(p cli.Params) (*v1alpha1.PipelineList, error) {
+func printPipelineDetails(out io.Writer, p cli.Params) error {
+
+	ps, prs, err := listPipelineDetails(p)
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to list pipelines from %s namespace\n", p.Namespace())
+		return err
+	}
+
+	if len(ps.Items) == 0 {
+		fmt.Fprintln(out, emptyMsg)
+		return nil
+	}
+
+	w := tabwriter.NewWriter(out, 0, 5, 3, ' ', tabwriter.TabIndent)
+	defer w.Flush()
+
+	fmt.Fprintln(w, header)
+
+	for _, pipeline := range ps.Items {
+		pr, ok := prs[pipeline.Name]
+		if !ok {
+			fmt.Fprintf(w, body,
+				pipeline.Name,
+				formatted.Age(pipeline.CreationTimestamp, p.Time()),
+				"---",
+				"---",
+				"---",
+				"---",
+			)
+			continue
+		}
+
+		fmt.Fprintf(w, body,
+			pipeline.Name,
+			formatted.Age(pipeline.CreationTimestamp, p.Time()),
+			pr.Name,
+			formatted.Age(*pr.Status.StartTime, p.Time()),
+			formatted.Duration(pr.Status.StartTime, pr.Status.CompletionTime),
+			formatted.Condition(pr.Status.Conditions[0]),
+		)
+
+	}
+	return nil
+}
+
+func printPipelineListObj(w io.Writer, p cli.Params, f *cliopts.PrintFlags) error {
+	ps, err := listAllPipelines(p)
+
+	if err != nil {
+		return err
+	}
+	return printer.PrintObject(w, ps, f)
+}
+
+func listAllPipelines(p cli.Params) (*v1alpha1.PipelineList, error) {
 	cs, err := p.Clientset()
 	if err != nil {
 		return nil, err
 	}
 
 	c := cs.TektonV1alpha1().Pipelines(p.Namespace())
-	return c.List(v1.ListOptions{})
+
+	ps, err := c.List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	// NOTE: this is required for -o json|yaml to work properly since
+	// tektoncd go client fails to set these; probably a bug
+	ps.GetObjectKind().SetGroupVersionKind(
+		schema.GroupVersionKind{
+			Version: "tekton.dev/v1alpha1",
+			Kind:    "PipelineList",
+		})
+	return ps, nil
+}
+
+type pipelineruns map[string]*v1alpha1.PipelineRun
+
+func listPipelineDetails(p cli.Params) (*v1alpha1.PipelineList, pipelineruns, error) {
+	cs, err := p.Clientset()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ps, err := listAllPipelines(p)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if len(ps.Items) == 0 {
+		return ps, pipelineruns{}, nil
+	}
+
+	runs, err := cs.TektonV1alpha1().
+		PipelineRuns(p.Namespace()).
+		List(metav1.ListOptions{})
+
+	if err != nil {
+		// TODO: may be just the pipeline details can be print
```

Comment:
good point  :+1: (may be an additional flag could help here?)

Reply:
may be better  to return a `specific error` and then check for that and just print the `pipelines`. Can submit that in a different PR.

Reply:
Lets go ahead with the current implementation and modify it when we really feel the need. 

I am marking it as resolved. Please reopen if you disagree